src/containers/User/SessionChecker/hooks.js

```diff
@@ -0,0 +1,16 @@
+import { useState, useEffect } from "react";
+import { useLoaderContext } from "src/Appv2/Loader";
+
+export function useSessionChecker() {
+  const [sessionExpired, setSessionExpired] = useState(false);
+  const { currentUser } = useLoaderContext();
+  useEffect(() => {
+    if (currentUser) {
+      const { sessionmaxage, lastlogintime } = currentUser;
+      const expired = sessionmaxage < Date.now() / 1000 - lastlogintime;
+      setSessionExpired(expired);
+    }
+  }, [currentUser]);
```

Comment:
Let's not add any dep for this effect so we run it on every rerender. After testing I realize the modal isn't showing up sometimes because of this.
```suggestion
  });
```